### PR TITLE
feat(install): added flag to pass ingress-class to exposecontroller

### PR DESCRIPTION
**Pull request by miguelgarcia**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

Add ingress-class flag to install command to pass the value to exposecontroller (https://github.com/jenkins-x/exposecontroller/commit/f5d8bd613d524b1d0da1ab2aad8258f371e384ff)

#### Special notes for the reviewer(s)

Not sure if I should also add IngressClass handling to pkg/kube/ingress.go

### CODE REVIEW

File: pkg/config/helm_values.go [hunk 1]
```diff
@@ -8,12 +8,13 @@ import (
...
-	HTTP        string `json:"http,omitempty"`
-	TLSAcme     string `json:"tlsacme,omitempty"`
-	PathMode    string `json:"pathMode,omitempty"`
-	UrlTemplate string `json:"urltemplate,omitempty"`
+	Domain        string `json:"domain,omitempty"`
+	Exposer       string `json:"exposer,omitempty"`
+	HTTP          string `json:"http,omitempty"`
+	TLSAcme       string `json:"tlsacme,omitempty"`
+	PathMode      string `json:"pathMode,omitempty"`
+	UrlTemplate   string `json:"urltemplate,omitempty"`
```
[the remark this answers is not in the document]
Comment (miguelgarcia) [reply pkg/config/helm_values.go]: It's out of the scope of this PR

**Comment by ccojocar** [comment]
@miguelgarcia Is this still WIP? Thanks

**Comment by miguelgarcia** [comment]
> @miguelgarcia Is this still WIP? Thanks

Hello,
I was waiting for exposecontroller version in jenkins-x-platform requirements to be updated to at least v2.3.94 to do a complete test (this PR uses a feature available since that exposecontroller version).

@ccojocar do you know when will that happen ?

Regards

**Comment by ccojocar** [comment]
@miguelgarcia Thanks for the update. I sent this PR that you will be able to use the latest version after it gets merged. 

https://github.com/jenkins-x/jx/pull/3236

**Comment by miguelgarcia** [comment]
> @miguelgarcia Thanks for the update. I sent this PR that you will be able to use the latest version after it gets merged.
> 
> #3236

Thanks @ccojocar , I see exposecontroller version was there, but it is also here https://github.com/jenkins-x/jenkins-x-platform/blob/master/requirements.yaml and here https://github.com/jenkins-x/jenkins-x-versions/pull/66.

Do you know which one is used when I do `jx install` ? Is it installed by https://github.com/jenkins-x/jx/blob/master/pkg/expose/exposecontroller.go or by helm when installing jenkins-x-platform chart ?

**Comment by ccojocar** [comment]
It is 2.3.97 https://github.com/jenkins-x/jenkins-x-versions/blob/835f2bf3e78c352e6fe435697210c36bf6359914/charts/jenkins-x/exposecontroller.yml#L1

**Comment by miguelgarcia** [comment]
> It is 2.3.97 https://github.com/jenkins-x/jenkins-x-versions/blob/835f2bf3e78c352e6fe435697210c36bf6359914/charts/jenkins-x/exposecontroller.yml#L1

Thanks @ccojocar, but I think it is still 2.3.89. After installing jx I run `kubectl describe jobs expose` and I get

```
Name:           expose
Namespace:      jx
Selector:       controller-uid=36398234-4024-11e9-8b8c-12a0b0124504
Labels:         chart=expose-2.3.89
                component=jenkins-x-expose
                heritage=Tiller
                release=jenkins-x
Annotations:    helm.sh/hook: post-install,post-upgrade
                helm.sh/hook-delete-policy: hook-succeeded
Parallelism:    1
Completions:    1
Start Time:     Wed, 06 Mar 2019 12:26:28 -0300
Pods Statuses:  1 Running / 0 Succeeded / 3 Failed
Pod Template:
  Labels:           chart=expose-2.3.89
                    controller-uid=36398234-4024-11e9-8b8c-12a0b0124504
                    heritage=Tiller
                    job-name=expose
                    release=jenkins-x
  Annotations:      sidecar.istio.io/inject: false
  Service Account:  expose
  Containers:
   expose:
    Image:      jenkinsxio/exposecontroller:2.3.89
    Port:       <none>
    Host Port:  <none>
    Command:
      /exposecontroller
    Environment:
      KUBERNETES_NAMESPACE:   (v1:metadata.namespace)
    Mounts:                  <none>
  Volumes:                   <none>
```

I think it's because of https://github.com/jenkins-x/jenkins-x-platform/blob/0e10af5320572e2b9911ba372cc01d03f1b9b315/requirements.yaml#L9

**Comment by rawlingsj** [comment]
while we move releases to new infra
/hold
/override serverless-jenkins

**Comment by miguelgarcia** [comment]
Hey @rawlingsj I also invite you to check this related issue https://github.com/jenkins-x/jenkins-x-platform/issues/4978 I've been trying to figure out why expose controller version is fixed to an old release there, since this PR depends on 2.3.94. I appreciate any help/hint . Thanks!

**Comment by rawlingsj** [comment]
> Hey @rawlingsj I also invite you to check this related issue jenkins-x/jenkins-x-platform#4978 I've been trying to figure out why expose controller version is fixed to an old release there, since this PR depends on 2.3.94. I appreciate any help/hint . Thanks! 


Ah, looks like we've not setup the updatebot config to auto upgrade the platform repo when we release exposecontroller, I'll sort that!

**Comment by miguelgarcia** [comment]
> > Hey @rawlingsj I also invite you to check this related issue [jenkins-x/jenkins-x-platform#4978](https://github.com/jenkins-x/jenkins-x-platform/issues/4978) I've been trying to figure out why expose controller version is fixed to an old release there, since this PR depends on 2.3.94. I appreciate any help/hint . Thanks!
> 
> Ah, looks like we've not setup the updatebot config to auto upgrade the platform repo when we release exposecontroller, I'll sort that!

Thanks ! I'll check this PR again when the updatebot updates exposecontroller to the new version

**Comment by miguelgarcia** [comment]
> > Hey @rawlingsj I also invite you to check this related issue [jenkins-x/jenkins-x-platform#4978](https://github.com/jenkins-x/jenkins-x-platform/issues/4978) I've been trying to figure out why expose controller version is fixed to an old release there, since this PR depends on 2.3.94. I appreciate any help/hint . Thanks!
> 
> Ah, looks like we've not setup the updatebot config to auto upgrade the platform repo when we release exposecontroller, I'll sort that!

Hey James, were you able to check this ? Let me know how may I help you with the updatebot

Regards!

**Comment by hferentschik** [comment]
@miguelgarcia, any chance you could sign-off your commit? I pushed a commit which fixed a Go format issue, but now the recently enabled DCO check fails on your commit.

**Pull request merged**